src/librustc/lint/context.rs

```diff
@@ -194,11 +198,24 @@ impl LintStore {
     pub fn register_future_incompatible(&mut self,
                                         sess: Option<&Session>,
                                         lints: Vec<FutureIncompatibleInfo>) {
-        let ids = lints.iter().map(|f| f.id).collect();
-        self.register_group(sess, false, "future_incompatible", ids);
-        for info in lints {
-            self.future_incompatible.insert(info.id, info);
+
+        for epoch in config::ALL_EPOCHS {
```

Comment:
comment plz

---

"Create a lint group for each epoch corresponding to the future compatibility warnings targeting that epoch"

---

(Do you think that this ought to include the lints for "all previous epochs", since they are additive?)

(Also, are these lint groups insta-stable?)

Reply:
We could. I think having them separate is cleaner, Rust doesn't do well with overlapping lint groups.

Rust lint stability is a very low bar, you just need to not remove the lint (aside from `register_removed_lint()` or whatever it's called). I think it's fine to insta-stable lint groups.